utils/awsutils/session.go

```diff
@@ -64,16 +64,36 @@ func CreateSession(config *SessionConfig) *session.Session {
 	}))
 }
 
-func NewSessionConfig(destinationConfig interface{}, timeout time.Duration, serviceName string) (*SessionConfig, error) {
+func NewSessionConfig(destinationConfig map[string]interface{}, timeout time.Duration, serviceName string) (*SessionConfig, error) {
+	if destinationConfig == nil {
+		return nil, errors.New("destinationConfig should not be nil")
+	}
 	sessionConfig := SessionConfig{}
 
-	jsonConfig, err := json.Marshal(destinationConfig)
-	if err != nil {
-		return nil, fmt.Errorf("[%s] Error while marshalling destination config :: %w", serviceName, err)
+	if region, ok := destinationConfig["Region"].(string); ok {
```

Comment:
Would this (with some tweaks) be a more elegant way of doing the same?
https://stackoverflow.com/a/26746461/3967127

It's better to avoid code like this.

Reply:
Updated the code to use mapstructure. Initially thought about using it but I have been told that we avoid using open sources unless absolutely needed so written manually, but mapstructure simplifies this process.